### DISCUSSION TITLE
nginx question: using alias will have 403 Permission denied, root won't and all permissions are set properly.

### DISCUSSION THREAD

**Discussion by qiulang** [Homebrew/discussions#4889] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: bf8991f0d777fe8e33105f0cd7bf1e5993eed60a
Last commit: 11 days ago
Core tap JSON: 31 Oct 15:11 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
...

  /usr/local/include/node/v8-weak-callback-info.h
  /usr/local/include/node/v8.h
  /usr/local/include/node/v8config.h
  /usr/local/include/node/zconf.h
  /usr/local/include/node/zlib.h

Warning: /usr/bin occurs before /opt/homebrew/bin in your PATH.
This means that system-provided programs will be used instead of those
provided by Homebrew. Consider setting your PATH so that
/opt/homebrew/bin occurs before /usr/bin. Here is a one-liner:
  echo 'export PATH="/opt/homebrew/bin:$PATH"' >> ~/.zshrc

The following tools exist at both paths:
  openssl

Warning: Homebrew's "bin" was not found in your PATH.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/bin:$PATH"' >> ~/.zshrc
```

### Description of issue

A simple nginx.conf to demonstrate the problem, the directory's permissions are set properly. Actually, I even run `chmod 777 to the folder` but I still got 403 error for `alias` directive

If I use root directive everything works fine, 403 only happens to alias  directive.

Why is that ?

        location / {
            root /Users/langqiu/some-project/dist;
            #alias /Users/langqiu/some-project/dist;
        }

**Comment by SMillerDev** [comment]
https://nginx.org/en/support.html

**Reply by qiulang** [reply, marked as answer]
OK, your reply makes me realize this is probably not a homebrew compile problem but a Nginx thing. So I finally found an answer here https://stackoverflow.com/questions/35259014/forbidden-location-when-using-alias-in-nginx-for-relative-urls . That is exactly the problem.

> The location and alias should both have a trailing / or neither have a trailing /
